src/components/views/rooms/EventTile.tsx

```diff
@@ -514,6 +514,13 @@ export default class EventTile extends React.Component<IProps, IState> {
 
         const room = MatrixClientPeg.get().getRoom(this.props.mxEvent.getRoomId());
         room?.on(ThreadEvent.New, this.onNewThread);
+
+        if (this.state.threadLastReply) {
+            if (this.state.threadLastReply.isEncrypted()) {
+                this.state.threadLastReply.once(MatrixEventEvent.Decrypted, this.onEventDecryption);
+                MatrixClientPeg.get().decryptEventIfNeeded(this.state.threadLastReply);
```

Comment:
This handler may never fire (and thus leak a bind reference) given that `decryptEventIfNeeded` does a bunch of short-circuits.

Can we use `decryptEventIfNeeded(...).then(this.onEventDecryption);`

Reply:
Hm but that event listener will be removed when the component unmount or when `threadLastReply` changes.
Happy to go with `.then()` but that will be impossible cancel the callback. If the component gets unmounted, and the event gets decrypted 1 second after we'll call `forceUpdate` on an unmounted component